odbc/tests/power_bi_tests.rs

```diff
@@ -0,0 +1,80 @@
+extern crate odbc_sys;
+use odbc_sys::*;
+use std::ops::FnOnce;
+use std::ptr::null_mut;
+
+/// Validate that the outcome is SQL_SUCCESS if validate_success is true.
+/// Otherwise, only stops with an exception on SQL_ERROR.
+fn execute_odbc_call<F>(validate_success: bool, func: F)
+where
+    F: FnOnce() -> SqlReturn,
+{
+    let outcome: SqlReturn = func();
+    if validate_success {
+        assert_eq!(SqlReturn::SUCCESS, outcome)
+    } else if SqlReturn::ERROR == outcome {
+        panic!("ODBC call failed. Outcome is SQL_ERROR")
+        // TODO : Get the error information for the diagnostic records
+    }
+}
```

Comment:
this function seems to have some sharp edges, and doesn't seem like it will be cleanly extensible to testing other SQL return codes. Is there any reason we can't just use 
```
assert_eq!(
    SqlReturn::SUCCESS,
    SQLAllocHandle(...)
);
```
in the places where `execute_odbc_call` is currently used? It doesn't seem any less expressive/compact, and it's a good deal more explicit. You could even add convenience macros like `assert_success!` if that was a common pattern that you wanted to make more terse
```suggestion
```